src/Startup.cpp

```diff
@@ -377,6 +378,11 @@ Startup(UI::Display &display)
   // Read the terrain file
   main_window->LoadTerrain();
 
+  // Show onboarding dialog
+  if (true || Profile::GetPath(ProfileKeys::MapFile) == nullptr) {
+    dlgOnboardingShowModal();
+  }
+
```

Comment:
Please use a own ProfileKey. Depending on something else is in transparent. We already have a lot of "hidden" behaviors  in XCSoar. Lets try to not to add.